### ISSUE TITLE
Adding stylesheet prefetch type in HtmlResourceHint

### ISSUE THREAD

**Issue by rustyconover** [assetgraph/assetgraph#1138]
Hi,

When parsing

https://www.instagram.com/accounts/login/

`<link rel="prefetch" as="stylesheet" href="/static/bundles/es6/FeedPageContainer.css/263e0ec404b3.css" type="text/css" crossorigin="anonymous">`

Encountering that tag that causes an error since `stylesheet` isn't in `validRequestDestinations` which is sourced from https://www.w3.org/TR/preload/#as-attribute.

Would it be possible to add `stylesheet` as an alias for `style`?

Rusty

**Comment by papandreou** [comment]
Do browsers support `stylesheet` as an alias? If they don't, we probably shouldn't alias it in assetgraph either.

However, what is "an error"? If it's breaking assetgraph, we should of course fix that.

**Comment by papandreou** [comment]
Ah, yes it is:

```
/Users/andreaslind/work/assetgraph/lib/relations/Html/HtmlResourceHint.js:194
      throw new Error(
      ^

Error: relations/HtmlResourceHint: Not a valid type for the as-method: stylesheet
    at HtmlPrefetchLink.set as [as as] (/Users/andreaslind/work/assetgraph/lib/relations/Html/HtmlResourceHint.js:194:13)
    at extendDefined (/Users/andreaslind/work/assetgraph/lib/util/extendDefined.js:7:21)
    at new Relation (/Users/andreaslind/work/assetgraph/lib/relations/Relation.js:49:5)
    at new HtmlRelation (/Users/andreaslind/work/assetgraph/lib/relations/HtmlRelation.js:8:1)
    at new HtmlResourceHint (/Users/andreaslind/work/assetgraph/lib/relations/Html/HtmlResourceHint.js:18:1)
    at new HtmlPrefetchLink (/Users/andreaslind/work/assetgraph/lib/relations/Html/HtmlPrefetchLink.js:8:1)
    at Html._vivifyRelation (/Users/andreaslind/work/assetgraph/lib/assets/Asset.js:1057:26)
    at _outgoingRelations.findOutgoingRelationsInParseTree.map (/Users/andreaslind/work/assetgraph/lib/assets/Asset.js:1088:36)
    at Array.map (<anonymous>)
    at Html.get outgoingRelations [as outgoingRelations] (/Users/andreaslind/work/assetgraph/lib/assets/Asset.js:1087:73)
    at gatherExternalRelations (/Users/andreaslind/work/assetgraph/lib/assets/Asset.js:1160:44)
    at Html.get externalRelations [as externalRelations] (/Users/andreaslind/work/assetgraph/lib/assets/Asset.js:1167:7)
    at Html.set url [as url] (/Users/andreaslind/work/assetgraph/lib/assets/Asset.js:1002:37)
    at Html.externalize (/Users/andreaslind/work/assetgraph/lib/assets/Asset.js:869:14)
    at AssetGraph.addAsset (/Users/andreaslind/work/assetgraph/lib/AssetGraph.js:301:19)
    at Object.<anonymous> (/Users/andreaslind/work/assetgraph/foo.js:5:22)
```

**Comment by papandreou** [comment]
Fixed in 6.4.2.
